### PR TITLE
Set strategy reminder

### PR DESCRIPTION
**Pull request by andela-oesho**
Set up email reminder for strategies
#358 fix similar code issue

### CODE REVIEW

File: app/mailers/notification_mailer.rb [hunk 1]
```diff
@@ -17,6 +17,20 @@ def refill_medication(reminder)
          subject: "Your refill for #{@medication.name} is coming up soon!")
   end
 
+  def perform_strategy(reminder)
+    @strategy = reminder.strategy
+    @user = @strategy.user
+    mail(to: @user.email,
+         subject: "Don't forget to perform #{@strategy.name}!")
```
Comment (julianguyen) [inline app/mailers/notification_mailer.rb]: This should be in `en.yml`

File: app/mailers/notification_mailer.rb [hunk 2]
```diff
@@ -17,6 +17,20 @@ def refill_medication(reminder)
...
+  end
+
+  def weekly_self_care(reminder)
+    @strategy = reminder.strategy
+    @user = @strategy.user
+    mail(to: @user.email,
+         subject: "Done self_care #{@strategy.name} this week?")
```
Comment (julianguyen) [inline app/mailers/notification_mailer.rb]: This too should be in `en.yml`

File: app/mailers/notification_mailer.rb [hunk 3]
```diff
@@ -17,6 +17,20 @@ def refill_medication(reminder)
          subject: "Your refill for #{@medication.name} is coming up soon!")
```
Comment (julianguyen) [inline app/mailers/notification_mailer.rb]: This should be in `en.yml`

File: app/views/notification_mailer/perform_strategy.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<p>
+  Hi <%= @user.name %>,
```
Comment (julianguyen) [inline app/views/notification_mailer/perform_strategy.html.erb]: This text should be in `en.yml`

File: spec/factories.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
 FactoryGirl.define do
+  factory :strategy_email_reminder do
+
```
Comment (julianguyen) [inline spec/factories.rb]: Is this necessary since it's empty?

File: app/views/notification_mailer/perform_strategy.html.erb (hunk 4 above)
Comment (andela-oesho) [inline app/views/notification_mailer/perform_strategy.html.erb]: noted

File: app/views/notification_mailer/weekly_self_care.html.erb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+Hi <%= @user.name %>,
```
Comment (andela-oesho) [inline app/views/notification_mailer/weekly_self_care.html.erb]: noted

File: spec/javascripts/helpers/jasmine-jquery.js [hunk 7]
```diff
@@ -0,0 +1,841 @@
+/*!
```
Comment (julianguyen) [inline spec/javascripts/helpers/jasmine-jquery.js]: This jasmine-jquery library has already been merged! https://github.com/julianguyen/ifme/commit/1bcb1bf21ccfef9e44d74e015ef791b52cfbd2d2

File: spec/factories.rb (hunk 5 above)
Comment (andela-oesho) [inline spec/factories.rb]: yeah thats true will remove it, it was auto generated when i did `rails g model`

File: spec/javascripts/helpers/jasmine-jquery.js (hunk 7 above)
Comment (andela-oesho) [inline spec/javascripts/helpers/jasmine-jquery.js]: alright will squash dat commit

**Comment by julianguyen** [comment]
LGTM. Please take a look at the comments before you're ready to merge!

File: app/mailers/notification_mailer.rb (hunk 3 above)
Comment (andela-oesho) [inline app/mailers/notification_mailer.rb]: noted

File: app/mailers/notification_mailer.rb (hunk 2 above)
Comment (andela-oesho) [inline app/mailers/notification_mailer.rb]: noted

File: Gemfile.lock [hunk 8]
```diff
@@ -58,6 +60,9 @@ GEM
       rack (>= 0.9.0)
     bootstrap-datepicker-rails (1.6.1.1)
       railties (>= 3.0)
+    bootstrap-sass (3.3.7)
```
Comment (julianguyen) [inline Gemfile.lock]: What is bootstrap sass being used for?

File: app/assets/stylesheets/application.scss [hunk 9]
```diff
@@ -612,6 +614,11 @@ label {
   display: inherit;
   font-size: $font30;
   margin-bottom: 20px;
+  padding: 0;
+  font-weight: bold;
+  line-height: 1;
+  color: inherit ;
+  border-radius: .25em;
```
Comment (julianguyen) [inline app/assets/stylesheets/application.scss]: Can we use the equivalent px value?

File: app/assets/stylesheets/application.scss [hunk 10]
```diff
@@ -836,6 +843,11 @@ div.table div.table_cell {
     }
   }
 
+  td.active.day {
+    background: #6D0839 !important;
```
Comment (julianguyen) [inline app/assets/stylesheets/application.scss]: These colours should be in colors.scss

**Comment by julianguyen** [comment]
Added more comments!

File: circle.yml [hunk 11]
```diff
@@ -0,0 +1,31 @@
+## Customize the test machine
```
Comment (julianguyen) [inline circle.yml]: You can remove this from your PR! I pushed a config file to master https://github.com/julianguyen/ifme/commit/8c7c429244b331eb5af1b276c1c01e7f23cd9d27

File: app/assets/stylesheets/application.scss (hunk 9 above)
Comment (andela-oesho) [inline app/assets/stylesheets/application.scss]: sure will change that

File: app/assets/stylesheets/application.scss (hunk 10 above)
Comment (andela-oesho) [inline app/assets/stylesheets/application.scss]: noted

File: circle.yml (hunk 11 above)
Comment (andela-oesho) [inline circle.yml]: yeah i just saw it after pulling from master thanks

**Comment by andela-oesho** [comment]
#358 fixed

**Comment by julianguyen** [comment]
LGTM! Thanks a bunch @andela-oesho for working on this 💯

File: db/schema.rb [hunk 12]
```diff
@@ -142,6 +142,21 @@
     t.boolean  "active",        null: false
     t.datetime "created_at"
     t.datetime "updated_at"
+    t.integer  "strategy_id"
+  end
+
+  create_table "self_care_reminders", force: :cascade do |t|
```
Comment (julianguyen) [inline db/schema.rb]: Why are there two tables with the same name `self_care_reminders` and `selfcare_reminders` and are these tables even being used?

File: db/schema.rb [hunk 13]
```diff
@@ -153,6 +168,29 @@
     t.datetime "created_at"
     t.datetime "updated_at"
     t.string   "name"
+    t.string   "self_care_strategy"
+    t.boolean  "daily_reminder_email"
+  end
+
+  create_table "strategy_email_reminders", force: :cascade do |t|
```
Comment (julianguyen) [inline db/schema.rb]: Where are these tables being used? Also the migration files are missing

File: config/initializers/devise.rb [hunk 14]
```diff
@@ -6,7 +6,7 @@
   # confirmation, reset password and unlock tokens in the database.
   # config.secret_key = 'e3296ce3d721a0debddfa1103bcb0d6e0b24e2fdbe44635ab1ccb0ee12ddc3f037ab11ebc1232d208fc4faf55c22d9696f02408a984462201eed890bdb9dc249'
 
-  config.secret_key = ENV['DEVISE_SECRET_KEY']
+  config.secret_key = ENV['DEVISE_SECRET_KEY']
```
Comment (julianguyen) [inline config/initializers/devise.rb]: This commit is unnecessary

File: db/schema.rb (hunk 13 above)
Comment (andela-oesho) [inline db/schema.rb]: i actually deleted dose tables, m shocked the migration files r missing while i have them here i will pull frm master, create just the needed tables and do a fresh pr

File: config/initializers/devise.rb (hunk 14 above)
Comment (andela-oesho) [inline config/initializers/devise.rb]: this was when i was trying to setup circle ci, thought i squashed the commit

**Pull request merged**